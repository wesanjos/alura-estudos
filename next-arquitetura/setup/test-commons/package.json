{
  "name": "@alura-estudo/test-commons",
  "version": "1.0.0",
  "private": true,
  "main": "index.js",
  "files": [
    "base.ts"
  ],
  "devDependencies": {
    "@testing-library/react": "14.0.0",
    "jest-styled-components": "7.1.1",
    "jest": "27.0.0",
    "ts-node": "10.7.0",
    "babel-jest": "28.0.3",
    "@babel/core": "7.17.10",
    "@babel/preset-env": "7.17.10",
    "@babel/preset-react": "7.16.7",
    "@babel/preset-typescript": "7.16.7",
    "@types/jest": "29.5.1"
  },
  "scripts": {
    "test": "echo 'Not implemented'",
    "lint": "echo 'Not implemented'"
  }
}
