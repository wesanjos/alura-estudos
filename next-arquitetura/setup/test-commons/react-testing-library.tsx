import { render as rtlRender } from "@testing-library/react";

function DefaultWrapper({ children }) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  return (ui, options = {}) => {
    return rtlRender(ui, { wrapper: AllTheProviders, ...options });
  };
};

export * from "@testing-library/react";

export const render = () =>
  console.error("Please create your own render method using ");
