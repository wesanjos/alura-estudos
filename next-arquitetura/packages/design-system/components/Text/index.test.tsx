import { Text } from "./index";
import { customRender } from "@alura-estudo/test-commons/react-testing-library";

const render = customRender();

describe("<Text/>", () => {
  it("renders h1 tag", () => {
    const { container } = render(<Text tag="h1">Sample Text</Text>);

    expect(container).toMatchSnapshot();
  });
});
