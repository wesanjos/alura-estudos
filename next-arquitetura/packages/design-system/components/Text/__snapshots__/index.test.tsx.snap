// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> renders h1 tag 1`] = `
<div>
  <h1
    class="sc-beqWaB dAAPWL"
  >
    Sample Text
  </h1>
</div>
`;
